crates/resolver/src/caching_client.rs

```diff
@@ -291,53 +291,117 @@ where
                 };
 
             // take all answers. // TODO: following CNAMES?
-            let mut response = response.into_message();
-            let answers = response.take_answers();
-            let additionals = response.take_additionals();
-            let authorities = response.take_authorities();
+            let mut message = response.into_message();
 
             // set of names that still require resolution
             // TODO: this needs to be enhanced for SRV
             let mut found_name = false;
+            let mut found_cname_target = false;
+            let mut min_ttl = cname_ttl;
+
+            // Scan through all sections to determine what we found and calculate minimum TTL
+            // We need this first pass to decide our strategy: return complete message vs filter
+            for r in message.all_sections() {
+                // because this resolved potentially recursively, we want the min TTL from the chain
+                min_ttl = min_ttl.min(r.ttl());
+
+                // TODO: disable name validation with ResolverOpts? glibc feature...
+                // restrict to the RData type requested
+                if query.query_class() == r.dns_class() {
+                    // standard evaluation, it's an any type or it's the requested type and the search_name matches
+                    #[allow(clippy::suspicious_operation_groupings)]
```

Comment:
What is clippy objecting to here? Is there another reasonable formulation that doesn't require lint suppression?